core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/part/EnvelopePart.java

```diff
@@ -172,7 +172,7 @@ private void runSanityChecks() {
         assert checkNaNFree(timeDeltas) : "NaNs in timeDeltas";
         assert checkPositive(speeds) : "negative speeds";
         assert checkPositive(timeDeltas) : "negative timeDeltas";
-        assert checkNonZero(timeDeltas) : "zero timeDeltas";
```

Comment:
We often need an epsilon-long step, and because the time deltas are much smaller than the position deltas we often have zeros there. The constraint has been moved to position instead to still prevent empty steps. (this is explained in the commit message)